pkg/odo/cli/service/operator_backend.go

```diff
@@ -261,5 +267,85 @@ func (b *OperatorBackend) buildCRDfromParams(o *CreateOptions, csv olm.ClusterSe
 		return nil, fmt.Errorf("the %q resource doesn't exist in specified %q operator", b.CustomResource, o.ServiceType)
 	}
 
-	return service.BuildCRDFromParams(cr, o.ParametersMap)
+	return svc.BuildCRDFromParams(cr, o.ParametersMap)
+}
+
+func (b *OperatorBackend) DescribeService(o *DescribeOptions, serviceName, app string) error {
+
+	clusterList, _, err := svc.ListOperatorServices(o.KClient)
+	if err != nil {
+		return err
+	}
+	var clusterFound *unstructured.Unstructured
+	for i, clusterInstance := range clusterList {
+		fullName := strings.Join([]string{clusterInstance.GetKind(), clusterInstance.GetName()}, "/")
+		if fullName == serviceName {
+			clusterFound = &clusterList[i]
+			break
+		}
+	}
+
+	devfileList, err := svc.ListDevfileServices(o.EnvSpecificInfo.GetDevfileObj())
+	if err != nil {
+		return err
+	}
+	devfileService, inDevfile := devfileList[serviceName]
+
+	item := NewServiceItem(serviceName)
+	item.InDevfile = inDevfile
+	item.Deployed = clusterFound != nil
+	if item.Deployed {
+		item.Manifest = clusterFound.Object
+	} else if item.InDevfile {
+		item.Manifest = devfileService.Object
+	}
```

Comment:
IIUC, if a deployed service is found, it will not take devfile service into account, correct?
And if that is the case, do you think we can write the code to not look for devfile service if it found a cluster service?
```suggestion
	item := NewServiceItem(serviceName)
	item.Deployed = clusterFound != nil
	if item.Deployed {
		item.Manifest = clusterFound.Object
	} else {
		devfileList, err := svc.ListDevfileServices(o.EnvSpecificInfo.GetDevfileObj())
		if err != nil {
			return err
		}
		devfileService, inDevfile := devfileList[serviceName]

		item.InDevfile = inDevfile
		if item.InDevfile {
			item.Manifest = devfileService.Object
		}
	}
```

Reply:
Yes, good catch, thanks

Reply:
Finally, no :( We need to populate the inDevfile field in the result, and so you always need to check if the service exists in the devfile

Reply:
Right right, that's a good catch :D